index.js

```diff
@@ -522,6 +528,31 @@
         });
       });
 
+  //# reduce :: Foldable b => (a -> b -> a) -> a -> b -> a
```

Comment:
This doesn't look quite right to me. In the Array case we take `[String]`, say, as `b`, but the function provided receives a value of type `String`—rather than `[String]`—as its `b`. I think we want:

```haskell
reduce :: Foldable f => (a -> b -> a) -> a -> f b -> a
```